src/internal/syntax/parser.rs

```diff
@@ -138,11 +138,11 @@ impl<'src> Parser<'src> {
     let token = self.current();
     if self.state.ignore_indent
       || token.is(Tok_Eof)
-      || previous.is(Tok_Semicolon)
-      || previous.is(Tok_SemicolonSemicolon)
       || matches!(token.ws, Some(n) if self.indent.is_eq(n))
     {
       Ok(())
+    } else if previous.is(Tok_Semicolon) || previous.is(Tok_SemicolonSemicolon) {
```

Comment:
If there's a semicolon, and the thing is on the next line, then it must still have equal indentation. If it's on the same line, then all is well.

Without this, stuff like
```python
if true: print x;
  print y
```
would work when it shouldn't have